web/src/pages/setting/SettingContent.vue

```diff
@@ -0,0 +1,98 @@
+<script lang="ts" setup>
+export interface LanguageWithIndex {
+  index: number
+}
+export interface LanguageWithIndex extends Language { }
+//LanguageWithIndex interface is merged by the Language type and custome index prop
+
+import { Language, languages } from "@/services/language"
+import {
+  Dialog,
+  DialogPanel,
+  DialogTitle,
+  DialogDescription,
+  TransitionChild,
+  TransitionRoot,
+  TabGroup,
+  TabList,
+  Tab,
+  TabPanels,
+  TabPanel,
+} from "@headlessui/vue"
+import LanguageSelector from "@/components/settings/LangSelector.vue"
+import { ref } from "vue"
+import { useI18n } from "vue-i18n"
+const { t } = useI18n()
+
+interface props {
+  show: boolean
+}
+const props = defineProps<props>()
+
+const categories = ref({
+  audioLanguage: languages,
+  subtitleLanguage: languages,
+})
+const curretnCategory = ref(t("audioLanguage")) //inital value of the tab
+
+const emits = defineEmits(["closeDialog"])
```

Comment:
If you need to emit something, use the convention:

defineEmits<{
(event: "{key}", value: Type): void // or if you need something
}>()

If you don't assign it to a variable, the same functionality is available in template with $emit("{event}", value)